src/components/PopupWithForm.js

```diff
@@ -5,6 +5,8 @@ class PopupWithForm extends Popup {
     this._handleFormSubmit = handleFormSubmit;
     this._formSelector = this._popupSelector.querySelector(".modal__container");
     this._inputList = this._formSelector.querySelectorAll(".modal__input");
+    this._button = this._formSelector.querySelector(".modal__save");
+    this._originalButtonText = this._button.textContent;
   }
 
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`